helix-view/src/document.rs

```diff
@@ -1584,16 +1584,48 @@ impl Document {
     /// Intialize/updates the differ for this document with a new base.
     pub fn set_diff_base(&mut self, diff_base: Vec<u8>) {
         if let Ok((diff_base, ..)) = from_reader(&mut diff_base.as_slice(), Some(self.encoding)) {
-            if let Some(differ) = &self.diff_handle {
-                differ.update_diff_base(diff_base);
-                return;
-            }
-            self.diff_handle = Some(DiffHandle::new(diff_base, self.text.clone()))
+            self.set_diff_base_raw(diff_base);
         } else {
             self.diff_handle = None;
         }
     }
 
+    pub fn set_diff_base_raw(&mut self, diff_base: Rope) {
+        if let Some(differ) = &self.diff_handle {
+            differ.update_diff_base(diff_base);
+            return;
+        }
+        self.diff_handle = Some(DiffHandle::new(diff_base, self.text.clone()))
+    }
+
+    pub fn update_diff_base(&mut self, path: &Path, diff_provider: &Git, diff_source: DiffSource) {
+        match diff_source {
+            DiffSource::Git => {
+                match diff_provider.get_diff_base(path) {
+                    Ok(diff_base) => self.set_diff_base(diff_base),
+                    Err(err) => {
+                        log::info!("{err:#?}");
+                        log::info!("failed to open diff base for for {}", path.display());
+                        self.diff_handle = None;
+                    }
+                }
+
+                self.version_control_head = match diff_provider.get_current_head_name(path) {
+                    Ok(res) => Some(res),
+                    Err(err) => {
+                        log::info!("{err:#?}");
+                        log::info!("failed to obtain current head name for {}", path.display());
+                        None
+                    }
+                };
+            }
+            DiffSource::File => {
+                self.set_diff_base_raw(self.original_text.clone());
+                self.version_control_head = None;
```

Comment:
I don't think version_control_head should be controlled by this setting, I think it should always use (git head). One we support multiple vcs this needs to be configured somehow but that should actually be automatic (a problem for the future). For now you can just hardcode git tough